sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsServiceSerializer.cs

```diff
@@ -0,0 +1,776 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal static class TextAnalyticsServiceSerializer
+    {
+        // TODO (pri 2): make the deserializer version resilient
+
+        #region Serialize Inputs
+
+        public static ReadOnlyMemory<byte> SerializeDetectLanguageInputs(IEnumerable<string> inputs, string countryHint)
+        {
+            var writer = new ArrayBufferWriter<byte>();
+            var json = new Utf8JsonWriter(writer);
+            json.WriteStartObject();
+            json.WriteStartArray("documents");
+            int id = 1;
+            foreach (var input in inputs)
+            {
+                json.WriteStartObject();
+                json.WriteString("countryHint", countryHint);
```

Comment:
You could get some perf gains by pre-encoding the names as `private static readonly` fields.

Reply:
@heaths Any reason to not use `const` fields?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8782